### PR TITLE
Ensure arrayField requires at least one element by default

### PR DESCRIPTION
**Pull request by teeeteee**

### CODE REVIEW

File: src/fields.tsx [hunk 1]
```diff
@@ -33,5 +33,5 @@ export function numberField(
 }
 
 export function arrayField<T>(props?: FieldProps<T[]>): FieldDefinition<T[]> {
-  return field({ default: [], ...props })
+  return field({ default: [], rules: [required, nonEmpty], ...props })
```
Comment (jcarver989) [inline src/fields.tsx:36]: Hmm I don't think we should assume that all use cases require a non-empty array. For example, it'd be fine for a multi-select style field to accept an empty array as valid input.

Instead we could add a `nonEmptyArrayField` convenience method, so we have both use-cases covered.

**Review by jcarver989 — CHANGES_REQUESTED** [review]
Nice Titi!

Changes look good. Just don't think we should make our `arrayField` helper enforce non-empty input. See comment about a suggestion.

File: src/fields.tsx (hunk 1 above)
Comment (teeeteee) [reply src/fields.tsx:36]: Makes sense 👍

**Review by jcarver989 — APPROVED** [review]
Thanks Titi!

**Pull request merged**